go/api/generic_commands.go

```diff
@@ -428,4 +428,89 @@ type GenericBaseCommands interface {
 	//
 	// [valkey.io]: https://valkey.io/commands/renamenx/
 	Renamenx(key string, newKey string) (Result[bool], error)
+
+	// Create a key associated with a value that is obtained by
+	// deserializing the provided serialized value (obtained via [valkey.io]: Https://valkey.io/commands/dump/).
+	//
+	// Parameters:
+	//  key - The key to create.
+	//	ttl - The expiry time (in milliseconds). If 0, the key will persist.
+	//  value - The serialized value to deserialize and assign to key.
+	//
+	// Return value:
+	//  Return OK if successfully create a key with a value </code>.
+	//
+	// Example:
+	// result, err := client.Restore("key",ttl, value)
+	//	if err != nil {
+	//	    // handle error
+	//	}
+	//	fmt.Println(result.Value()) // Output: OK
+	//
+	// [valkey.io]: https://valkey.io/commands/restore/
+	Restore(key string, ttl int64, value string) (Result[string], error)
+
+	// Create a key associated with a value that is obtained by
+	// deserializing the provided serialized value (obtained via [valkey.io]: Https://valkey.io/commands/dump/).
+	//
+	// Parameters:
+	//  key - The key to create.
+	//	ttl - The expiry time (in milliseconds). If 0, the key will persist.
+	//  value - The serialized value to deserialize and assign to key.
+	//  restoreOptions - Set restore options with replace and absolute TTL modifiers, object idletime and frequency
+	//
+	// Return value:
+	//  Return OK if successfully create a key with a value.
+	//
+	// Example:
+	// restoreOptions := api.NewRestoreOptionsBuilder().SetReplace().SetABSTTL().SetEviction(api.FREQ, 10)
+	// resultRestoreOpt, err := client.RestoreWithOptions(key, ttl, value, restoreOptions)
+	//	if err != nil {
+	//	    // handle error
+	//	}
+	//	fmt.Println(result.Value()) // Output: OK
+	//
+	// [valkey.io]: https://valkey.io/commands/restore/
+	RestoreWithOptions(key string, ttl int64, value string, option *RestoreOptions) (Result[string], error)
+
+	// Returns the internal encoding for the Valkey object stored at key.
+	//
+	// Note:
+	//  When in cluster mode, both key and newkey must map to the same hash slot.
+	//
+	// Parameters:
+	//  The key of the object to get the internal encoding of.
```

Comment:
```suggestion
	//  key -The key of the object to get the internal encoding of.
```

variable name missing here.